### ISSUE TITLE
"use strict" and comments

### ISSUE THREAD

**Issue by zertosh** [babel/babel#1030]
When there is a `'use strict'` at the top of a module, any comments before it get moved the bottom. I expect the comments and existing `'use strict'` to remain where they are.

**Original**
```
// comments
'use strict';

module.exports = {};
```

**Transformed**
```
"use strict";

module.exports = {};
// comments
```
---
When there isn't a `'use strict'`, the comments "stay in place" but the `'use strict'` gets inserted before them. This kinda messes up licensing/annotation docblocks. I expect the `'use strict'` to be added after the comments.

**Original**
```
// comments

module.exports = {};
```

**Transformed**
```
"use strict";

// comments

module.exports = {};
```

**Comment by stefanpenner** [comment]
this trolled us in ember, as we use comments to annotate files which should strip out use-strict (yes that sounds crazy, but jsc in ios 8 is seriously broken ...)

**Comment by sebmck** [comment]
Hm, so this issue is that when doing:

```javascript
// comments
'use strict';

module.exports = {};
```

The `// comments` comment gets attached to the use strict directive and then it gets destroyed and readded. Babel has then lost the node that the comment was attached to so it just shoves all undisplayed comments to the end of the file since it's better than nothing. Should be trivial to add back but this:

```javascript
// comments

module.exports = {};
```

is a bit more tricky. So what you want is for the use strict directive to be appended after all starting comments?

**Comment by sebmck** [comment]
Did it anyway since I figure that's what you want and it makes the most sense.

**Comment by zertosh** [comment]
perfectttttt!!!!! thanks @sebmck

**Comment by zertosh** [comment]
@sebmck Can you publish this fix? Thanks!

**Comment by sebmck** [comment]
Sorry for the delay, released as of 4.7.13. Thanks!
